python/cudf/cudf/_lib/parquet.pyx

```diff
@@ -87,3 +99,103 @@ cpdef read_parquet(filepath_or_buffer, columns=None, row_group=None,
         df.index.name = new_index_name
 
     return df
+
+cpdef write_parquet(
+        Table table,
+        path,
+        index=None,
+        compression=None,
+        statistics="ROWGROUP"):
+    """
+    Cython function to call into libcudf API, see `write_parquet`.
+
+    See Also
+    --------
+    cudf.io.parquet.write_parquet
+    """
+
+    # Create the write options
+    cdef string filepath = <string>str(path).encode()
+    cdef sink_info sink = sink_info(filepath)
+    cdef unique_ptr[table_metadata] tbl_meta = \
+        make_unique[table_metadata]()
+
+    cdef vector[string] column_names
+    cdef map[string, string] user_data
+
+    # Construct the column names
+    column_names.push_back(str.encode(table._index.name))
+    for col_name in table._column_names:
+        column_names.push_back(str.encode(col_name))
+
+    # Construct Pandas metadata
+    column_metadata = []
+    for c in table._column_names:
+        col = table._data[c]
+        column_metadata.append({
+            "name": c,
+            "field_name": c,
+            "pandas_type": pd.api.types.pandas_dtype(col.dtype).name,
+            "numpy_type": np.dtype(col.dtype).name,
+            "metadata": None
+        })
+
+    # Append the index column
```

Comment:
If index is a multiindex there would be multiple columns here. If we're not writing the index should this be written at all?

Reply:
Not 100% sure I get your drift here, but do understand multiindex would be problematic. Take a look at the last commit and see if that is what you are talking about.